File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/AddDocumentSnapshotListenerOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.Nullable;

public class AddDocumentSnapshotListenerOptions {

    private String reference;
    private final boolean includeMetadataChanges;
    private String callbackId;
    
    @Nullable
    private String databaseId;

    public AddDocumentSnapshotListenerOptions(String reference, @Nullable Boolean includeMetadataChanges, String callbackId, @Nullable String databaseId) {
        this.reference = reference;
        this.includeMetadataChanges = includeMetadataChanges == null ? false : includeMetadataChanges;
        this.callbackId = callbackId;
        this.databaseId = databaseId;
    }

    public String getReference() {
        return reference;
    }

    public boolean isIncludeMetadataChanges() {
        return includeMetadataChanges;
    }

    public String getCallbackId() {
        return callbackId;
    }
    
    @Nullable
    public String getDatabaseId() {
        return databaseId;
    }
}

File: packages/firestore/ios/Plugin/Classes/Options/AddDocumentSnapshotListenerOptions.swift
import Foundation

@objc public class AddDocumentSnapshotListenerOptions: NSObject {
    private var reference: String
    private var includeMetadataChanges: Bool
    private var callbackId: String
    private var databaseId: String?

    init(reference: String, includeMetadataChanges: Bool, callbackId: String, databaseId: String? = nil) {
        self.reference = reference
        self.includeMetadataChanges = includeMetadataChanges
        self.callbackId = callbackId
        self.databaseId = databaseId
    }

    func getReference() -> String {
        return reference
    }

    func getIncludeMetadataChanges() -> Bool {
        return includeMetadataChanges
    }

    func getCallbackId() -> String {
        return callbackId
    }

    func getDatabaseId() -> String? {
        return databaseId
    }
}
